lib/campaign/CampaignState.h

```diff
@@ -293,6 +293,8 @@ class DLL_LINKAGE CampaignState : public Campaign
 	static JsonNode crossoverSerialize(CGHeroInstance * hero);
 	static CGHeroInstance * crossoverDeserialize(const JsonNode & node, CMap * map);
 
+	std::string campaignSet = "";
```

Comment:
Non-trivial classes don't need zero-init.
```suggestion
	std::string campaignSet;
```